examples/tasks/advanced/run_fancy_app.py

```diff
@@ -33,28 +33,33 @@ class MyApplication(TaskApplication):
     # TaskApplication interface
     # -------------------------------------------------------------------------
 
-    app_name = Str("MyPyfaceApplication")
+    id = "MyPyfaceApplication"
 
-    window_size = Tuple((800, 600))
+    name = "Python Editor"
 
+    window_size = (800, 600)
 
-    def start(self):
-        starting = super(MyApplication, self).start()
-        if not starting:
-            return False
+    extra_actions = List(Instance(
+        'pyface.tasks.action.schema_addition.SchemaAddition'
+    ))
 
-        self.create_new_task_window()
-        return True
+    def _on_window_closing(self, window, trait, old, new):
+        """ Ask confirmation when a window is closed. """
+        from pyface.api import confirm, YES
+
+        msg = "Are you sure you want to close the window?"
+        return_code = confirm(None, msg)
+        if return_code != YES:
+            logger.debug("Window closing even was veto-ed")
+            new.veto = True
```

Comment:
Thanks! I had the same code to add to the example, and was waiting for your PR. +1